Application/Published/Upgrade-AdminApp/Deploy-EdFiOdsAdminApp.ps1

```diff
@@ -0,0 +1,220 @@
+
+	$sqlServer = @{
+		HostName = $hostName
+		AdminCredentials = (Get-CredentialFromConsole $adminCredentialsPrompt)	
+	}	
+	return $sqlServer
+}
+
+function Deploy-AdminApp($odsDeployInfo)
+{
+	$resourceGroupName = $odsDeployInfo.ResourceGroupName
+	$appInsightsLocation = $odsDeployInfo.AppInsightsLocation	
+	$sqlServer = $odsDeployInfo.SqlServerInfo
+	$productionApiUrl = $odsDeployInfo.ProductionApiUrl	
+
+	if(-not $EncryptionKey)
+	{
+		$aes = [System.Security.Cryptography.Aes]::Create()
+		$aes.KeySize = 256
+		$aes.GenerateKey()
+		$EncryptionKey = [System.Convert]::ToBase64String($aes.Key)
+	}
+	
+	$deployParameters = New-Object -TypeName Hashtable
+	$deployParameters.Add("version", $AdminAppVersion)
+	$deployParameters.Add("edition", $Edition)
+	$deployParameters.Add("appInsightsLocation", $appInsightsLocation)
+	$deployParameters.Add("odsInstanceName", $resourceGroupName)
+	$deployParameters.Add("sqlServerAdminLogin", $sqlServer.AdminCredentials.UserName)
+	$deployParameters.Add("sqlServerAdminPassword", $sqlServer.AdminCredentials.Password)
+	$deployParameters.Add("productionApiUrl", $productionApiUrl)	
+	$deployParameters.Add("metadataCacheTimeOut", $CacheTimeOut)
+	$deployParameters.Add("adminAppNameToDeploy", $AdminAppName)
+	$deployParameters.Add("encryptionKey", $EncryptionKey)
+	
+	$templateFile = $AdminAppTemplateFile
+	$templateParametersFile = $AdminAppTemplateParametersFile
+
+	Try
+	{
+		Write-Host "Deploying ODS Admin App"		
+		$deploymentResult = New-AzureRmResourceGroupDeployment -DeploymentDebugLogLevel All -Name ((Get-ChildItem $templateFile).BaseName + '-' + ((Get-Date).ToUniversalTime()).ToString('MMdd-HHmm')) `
+									   -ResourceGroupName $resourceGroupName `
+									   -TemplateFile $templateFile `
+									   -TemplateParameterFile $templateParametersFile `
+									   @deployParameters `
+									   -Force -Verbose -ErrorAction Stop
+	}
+	Catch
+	{
+		Write-Error $_.Exception.Message		
+	}
+
+	$adminAppUrl = $deploymentResult.Outputs.adminAppUrl.Value
+	Write-Success "Ed-Fi ODS Admin App accessible at $adminAppUrl"
+	
+	return $deploymentResult
+}
+
+function IsDBInstalled([string]$Server, [string]$database)
+{	 
+	$t=Invoke-Sqlcmd -ServerInstance $Server -Username  $user -Password  $pwd -Database "master" -Query "select 1 from sys.databases where name='$database'" -OutputSqlErrors $true 
+	if (!$t) {				
+		Write-Error "Failed to connect to [$database] database on [$Server]" -ErrorAction Stop
+	} else {				   
+		Write-Success "[$database] Database exists in SQL Server [$Server]"
+	}
+}
+
+function Run-DbMigrations($sqlServerInfo)
+{
+	$destinationZipFile = "$PSScriptRoot/Database.zip"
+	$scriptFilesPath = "$PSScriptRoot/ScriptFiles"
+	Invoke-WebRequest -Uri "https://odsassets.blob.core.windows.net/public/adminapp/Release/2.2.0/Edfi.suite3.ods.adminapp.database.zip" -OutFile $destinationZipFile
```

Comment:
What do you think about moving the `Uri` value to a constant at the top of the file? That will make this line shorter / easier to read, and it will be easier to find and update this value for the next release.

Also, should it use `$AdminAppVersion` instead of hard-coding `2.2.0`?